csharp/src/Google.Protobuf/WriteBufferHelper.cs

```diff
@@ -118,10 +119,12 @@ public static int GetSpaceLeft(ref WriterInternalState state)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void RefreshBuffer(ref Span<byte> buffer, ref WriterInternalState state)
+        public static void RefreshBuffer(ref Span<byte> buffer, ref WriterInternalState state, int sizeHint = 0)
         {
             if (state.writeBufferHelper.codedOutputStream?.InternalOutputStream != null)
             {
+                Debug.Assert(sizeHint == 0, "CodedOutputStream does not support sizeHint.");
```

Comment:
We (.NET team) use Debug.Assert to check for things that shouldn't happen, but aren't worth checking at runtime. I couldn't find any other uses of Debug.Assert outside of test code.

What do you want to do here:
1. Remove this?
2. Verify are runtime (check and throw an exception)
3. Ok to leave as is